packages/contracts-bedrock/src/universal/CrossDomainMessenger.sol

```diff
@@ -219,6 +223,10 @@ abstract contract CrossDomainMessenger is
         // On L2 this function should be a no-op, because paused will always return false.
         require(paused() == false, "CrossDomainMessenger: paused");
 
+        if (isCustomGasToken()) {
```

Comment:
We want this check in `sendMessage`. It technically doesn't have to be in this branch because if its enforced in send message then its enforced to be 0 on this side. You can see the `isOtherMessenger` branch to confirm

Reply:
I think its fine to leave here